### PR TITLE
Task back end

### PR DESCRIPTION
**Pull request by Dawid-Dwk**

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#Requests
```
Comment (mcwiekala) [inline API.md:1]: add postman request to get/create task

File: packages/backend/src/infrastructure/express/router/task.routes.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  constructor(app: express.Application) {
+    super(app, 'UsersRoutes')
+  }
+
+  // prettier-ignore
+  configureRoutes(): express.Application {
+    this.app
+    .route(`${V1}/`)
+    .get(
+        (req: express.Request, res: express.Response) => res.send('Helo World!')
```
Comment (mcwiekala) [inline packages/backend/src/infrastructure/express/router/task.routes.ts:17]: - request should be a DTO eg. CreateTaskDTO
- response should be a DTO eg. TaskDTO

Comment (mcwiekala) [inline packages/backend/src/infrastructure/express/router/task.routes.ts:17]: - request should be a DTO eg. CreateTaskDTO
- response should be a DTO eg. TaskDTO

File: packages/backend/src/modules/task/task.controller.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { CreateTaskCommand } from 'shared'
+import express from 'express'
+import _taskService from './task.service'
+
+class TaskController {
+  // eslint-disable-next-line class-methods-use-this
+  async createTask(req: express.Request, res: express.Response) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.controller.ts:7]: well it's TypeScript so please use types:
- request should be a DTO eg. CreateTaskDTO
- response should be a DTO eg. TaskDTO

Comment (mcwiekala) [inline packages/backend/src/modules/task/task.controller.ts:7]: well it's TypeScript so please use types:
- request should be a DTO eg. CreateTaskDTO
- response should be a DTO eg. TaskDTO

File: packages/backend/src/modules/task/task.controller.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    console.log(`Received new task with title: ${req.body.title}`)
+    const createTaskCommand: CreateTaskCommand = req.body
+    const newTask = await _taskService.createTask(createTaskCommand)
+    return res.status(200).send(newTask._id)
+  }
+  // eslint-disable-next-line class-methods-use-this
+  async getTasks(req: express.Request, res: express.Response) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.controller.ts:14]: well it's TypeScript so please use types:
- request should be a DTO eg. CreateTaskDTO
- response should be a DTO eg. TaskDTO

Comment (mcwiekala) [inline packages/backend/src/modules/task/task.controller.ts:14]: well it's TypeScript so please use types:
- request should be a DTO eg. CreateTaskDTO
- response should be a DTO eg. TaskDTO

File: packages/backend/src/modules/task/task.controller.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    const Tasks = await _taskService.getTasks()
+    return res.status(200).json(Tasks)
+  }
+  // eslint-disable-next-line class-methods-use-this
+  async getTask(req: express.Request, res: express.Response) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.controller.ts:19]: well it's TypeScript so please use types:
- request should be a DTO eg. CreateTaskDTO
- response should be a DTO eg. TaskDTO

Comment (mcwiekala) [inline packages/backend/src/modules/task/task.controller.ts:19]: well it's TypeScript so please use types
as above

File: packages/backend/src/modules/task/task.model.ts [hunk 6]
```diff
@@ -1,30 +1,28 @@
...
   ],
 
   comments: [
     {
-      _id: ObjectId,
       createdAt: { type: Date, default: Date.now },
       content: String,
       user: {
-        _id: ObjectId,
+        user_id: ObjectId,
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.model.ts:20]: leave it as it is? Why this change?

Comment (mcwiekala) [inline packages/backend/src/modules/task/task.model.ts:20]: leave it as it is? Why this change?

File: packages/backend/src/modules/task/task.model.ts [hunk 7]
```diff
@@ -1,30 +1,28 @@
...
 
 const taskSchema = new mongoose.Schema({
-  _id: ObjectId,
   title: String,
   description: String,
   imageCoverId: String,
   attachments: [
     {
-      _id: ObjectId
+      attachments_id: ObjectId
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.model.ts:11]: leave it as it is? Why this change?

File: packages/backend/src/modules/task/task.repository.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
+import { CreateTaskCommand } from 'shared'
+import TaskModel from './task.model'
+import BoardModel from '../board/Board'
+
+class TaskRepository {
+  private _taskModel = TaskModel
+
+  private _boardModel = BoardModel
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.repository.ts:8]: It's TaskRepository so why you use BoardModel here? Create BoardRepository and call it from BoardService

Comment (mcwiekala) [inline packages/backend/src/modules/task/task.repository.ts:8]: It's TaskRepository so why you use BoardModel here? Create BoardRepository and call it from BoardService

File: packages/backend/src/modules/task/task.repository.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    const columnIndex = board.columns.findIndex(
+      (x: { _id: string }) => String(x._id) === idCol.idColumn
+    )
+
+    board.columns[columnIndex].tasks.push(savedTask)
+    await board.save()
+    return savedTask
+  }
+
+  async getTasks() {
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.repository.ts:28]: change to findAll

File: packages/backend/src/modules/task/task.repository.ts [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+    const tasks = await this._taskModel.find()
+    console.log(tasks)
+    return tasks
+  }
+
+  async getTask(_id: string) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.repository.ts:34]: change to findById

Comment (mcwiekala) [inline packages/backend/src/modules/task/task.repository.ts:34]: change to findById

File: packages/backend/src/modules/task/task.service.ts [hunk 11]
```diff
@@ -0,0 +1,20 @@
+import { CreateTaskCommand } from 'shared'
+import _taskRepository from './task.repository'
+
+class TaskService {
+  // eslint-disable-next-line class-methods-use-this
+  createTask(createTaskCommand: CreateTaskCommand) {
+    console.log('Handling new task')
+    return _taskRepository.createTask(createTaskCommand)
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.service.ts:8]: use here _boardRepository to change data from board

Comment (mcwiekala) [inline packages/backend/src/modules/task/task.service.ts:8]: use here _boardRepository to change data from board

File: packages/backend/src/modules/task/task.service.ts [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  }
+  // eslint-disable-next-line class-methods-use-this
+  getTasks() {
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.service.ts:11]: add return value

File: packages/backend/src/modules/task/task.service.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    return _taskRepository.getTasks()
+  }
+  // eslint-disable-next-line class-methods-use-this
+  getTask(_id: string) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.service.ts:15]: add return value

File: packages/backend/src/modules/task/task.repository.ts (hunk 10 above)
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.repository.ts:34]: add return value

File: packages/backend/src/modules/task/task.repository.ts [hunk 14]
```diff
@@ -0,0 +1,41 @@
+import { CreateTaskCommand } from 'shared'
+import TaskModel from './task.model'
+import BoardModel from '../board/Board'
+
+class TaskRepository {
+  private _taskModel = TaskModel
+
+  private _boardModel = BoardModel
+
+  async createTask(createTaskCommand: CreateTaskCommand) {
```
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.repository.ts:10]: add return value

File: packages/backend/src/modules/task/task.controller.ts (hunk 5 above)
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.controller.ts:19]: add return value

File: packages/backend/src/modules/task/task.controller.ts (hunk 4 above)
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.controller.ts:14]: return type?

File: packages/backend/src/modules/task/task.controller.ts (hunk 3 above)
Comment (mcwiekala) [inline packages/backend/src/modules/task/task.controller.ts:7]: return type?

File: packages/frontend/src/components/Task/Task.tsx [hunk 15]
```diff
@@ -5,6 +5,7 @@ import { useState } from 'react'
 import UserType from '../../types/user'
```
Comment (mcwiekala) [inline packages/frontend/src/components/Task/Task.tsx:5]: remove frontend changes from this branch

File: packages/frontend/src/components/TaskModal/TaskModal.tsx [hunk 16]
```diff
@@ -17,7 +17,6 @@ import GrayButtonFilled from '../GrayButtonFilled/GrayButtonFilled'
```
Comment (mcwiekala) [inline packages/frontend/src/components/TaskModal/TaskModal.tsx:17]: remove frontend changes from this branch

File: packages/shared/src/api/dto/CreateTaskCommand.ts [hunk 17]
```diff
@@ -0,0 +1,7 @@
+export interface CreateTaskCommand {
```
Comment (mcwiekala) [inline packages/shared/src/api/dto/CreateTaskCommand.ts:1]: remove frontend changes from this branch

File: API.md (hunk 1 above)
Comment (Dawid-Dwk) [reply API.md:1]: added

File: packages/backend/src/modules/task/task.model.ts (hunk 6 above)
Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.model.ts:20]: previously it did not correctly generate mongoose IDs

Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.model.ts:20]: previously it did not correctly generate mongoose IDs

Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.model.ts:20]: previously it did not correctly generate mongoose IDs

File: packages/backend/src/modules/task/task.model.ts (hunk 7 above)
Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.model.ts:11]: previously it did not correctly generate mongoose IDs

File: packages/backend/src/modules/task/task.repository.ts (hunk 9 above)
Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.repository.ts:28]: Done

File: packages/backend/src/modules/task/task.repository.ts (hunk 10 above)
Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.repository.ts:34]: Done

File: packages/backend/src/modules/task/task.model.ts (hunk 6 above)
Comment (kamack38) [reply packages/backend/src/modules/task/task.model.ts:20]: isn't it better to use the mongo ID rather than creating a new one?

File: packages/backend/src/modules/task/task.service.ts (hunk 12 above)
Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.service.ts:11]: Done

File: packages/backend/src/modules/task/task.service.ts (hunk 13 above)
Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.service.ts:15]: Done

File: packages/backend/src/modules/task/task.repository.ts (hunk 10 above)
Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.repository.ts:34]: Done

File: packages/backend/src/modules/task/task.repository.ts (hunk 14 above)
Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.repository.ts:10]: Done

File: packages/backend/src/modules/task/task.service.ts (hunk 11 above)
Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.service.ts:8]: Done

Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.service.ts:8]: Done

File: packages/backend/src/modules/task/task.repository.ts (hunk 8 above)
Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.repository.ts:8]: Done

Comment (Dawid-Dwk) [reply packages/backend/src/modules/task/task.repository.ts:8]: Done

File: packages/frontend/src/components/Task/Task.tsx (hunk 15 above)
[the remark this answers is not in the document]
Comment (Dawid-Dwk) [reply packages/frontend/src/components/Task/Task.tsx]: Done

Comment (Dawid-Dwk) [reply packages/frontend/src/components/Task/Task.tsx]: Done

File: packages/frontend/src/components/TaskModal/TaskModal.tsx (hunk 16 above)
Comment (Dawid-Dwk) [reply packages/frontend/src/components/TaskModal/TaskModal.tsx]: Done

File: packages/shared/src/api/dto/CreateTaskCommand.ts (hunk 17 above)
Comment (Dawid-Dwk) [reply packages/shared/src/api/dto/CreateTaskCommand.ts]: Done

**Review by kamack38 — APPROVED** [review]
LGTM
